doc/langref.html.in

```diff
@@ -2180,16 +2180,20 @@ or
       Unlike normal structs, {#syntax#}packed{#endsyntax#} structs have guaranteed in-memory layout:
       </p>
       <ul>
-        <li>Fields remain in the order declared, least to most significant.</li>
+        <li>Fields are arranged in the order declared, from the least to the most significant bits of a backing integer.</li>
         <li>There is no padding between fields.</li>
-        <li>Zig supports arbitrary width {#link|Integers#} and although normally, integers with fewer
-        than 8 bits will still use 1 byte of memory, in packed structs, they use
-        exactly their bit width.
+        <li>The backing integer is subject to the same rules as any integer, including {#link|alignment|Alignment#}, 
+        having a maximum bit count of 65535, and the host endianness.
+        On a big endian system, the first declared field will have the highest memory address.
         </li>
+        <li>The backing integer has the same bit count as the fields' total bit width.</li> 
+        <li>Field access and assignment can be understood as shorthand for bitshifts on the backing integer.</li>
+        <li>Integers will use exactly as many bits as they have. For example, a {#syntax#}u5{#endsyntax#} will use 5 bits of the backing integer.
         <li>{#syntax#}bool{#endsyntax#} fields use exactly 1 bit.</li>
         <li>An {#link|enum#} field uses exactly the bit width of its integer tag type.</li>
         <li>A {#link|packed union#} field uses exactly the bit width of the union field with
         the largest bit width.</li>
+        <li>Packed structs, when they themselves are within a packed struct, will only use as many bits as their backing integer.</li>
```

Comment:
Can you add anything to this doc about backing ints that are not a multiple of 8 bits?  

(Perhaps this is mostly relevant in the MMIO case, where users wouldn't want undefined bits, and the load/store operations are going to be an 8-bit multiple?  Hrm, and would a u24 get rounded up to a u32?)

The PR is still a doc improvement without this detail, so feel free to leave it off.

Reply:
> Can you add anything to this doc about backing ints that are not a multiple of 8 bits?

This is intended to be captured by:
"The backing integer is subject to the same rules as any integer, including **alignment**,"

> (Perhaps this is mostly relevant in the MMIO case, where users wouldn't want undefined bits, and the load/store operations are going to be an 8-bit multiple? Hrm, and would a u24 get rounded up to a u32?)

I suspect you are correct about its relevance to the MMIO case, and just below in the docs it talks about using overaligned pointers to override the alignment of the packed struct, which I guess is what the MMIO use case needs to do to avoid writing/reading to memory areas that are just padding for alignment?

"Packed structs have the same alignment as their backing integer, however, overaligned pointers to packed structs can override this:"

Reply:
> Can you add anything to this doc about backing ints that are not a multiple of 8 bits?

In case you were referring to packed struct fields inside a packed struct field, 
https://github.com/ziglang/zig/pull/21741/commits/6c330beb093f4862647301d24608889b22a9a614
adds an example to to help explain this case.